tests/report_tests/plugin_tests/openstack/openstack.py

```diff
@@ -0,0 +1,162 @@
+                        "Unable to install requested packages "
+                        "dnf-plugins-core")
+            shell_out("dnf config-manager --enable crb")
+
+    def setup_mocked_packages(self):
+        this_distro = distro.detect()
+        if this_distro.name == "centos" or this_distro.name == "centos-stream":
+            if this_distro.version == "9":
+                self.osp_release = "yoga"
+            if this_distro.version == "8":
+                self.osp_release = "victoria"
+            self._prep_crb_repo()
+        if self.osp_release != "":
+            pkg_to_install = f"centos-release-openstack-{self.osp_release}"
+            installed = distro_packages.install_distro_packages(
+                {this_distro.name: [pkg_to_install]})
+            if not installed:
+                raise Exception(
+                    "Unable to install requested packages "
+                    f"{pkg_to_install}")
+            self._write_file_to_tmpdir(
+                'mocked_osp_package', f'["{pkg_to_install}"]')
+        super().setup_mocked_packages()
+
+    def teardown_mocked_packages(self):
+        this_distro = distro.detect()
+        if this_distro.name == "centos" or this_distro.name == "centos-stream":
+            pkgs = self.read_file_from_tmpdir('mocked_osp_package')
+            if not pkgs:
+                return
+            pkgs = json.loads(pkgs)
+            for pkg in pkgs:
+                self.sm.remove(pkg)
+        super().teardown_mocked_packages()
+
+    def test_cinder_conf_collected_and_scrubbed(self):
+        self.assertFileCollected('/etc/cinder/cinder.conf')
+
+        keys_to_mask = [
+            'cmB4zBYq3VWFMNqNKFLcqS5Zq8ystLLsTd5BFLbCtX67qShnhgHFxxRFjkhbY54x',
+            'wPhFqY69x94YVJc7STrVH3CfsFrrcZPYw8NS2pjhzqyzw7wrL2VnTmN58c5XTnfV',
+            'Ck3r7zf6B6PscfjWhhj2zJdy8SNXYd59',
+        ]
+        for key in keys_to_mask:
+            self.assertFileNotHasContent(
+                '/etc/cinder/cinder.conf', key)
+
+    def test_glance_conf_collected_and_scrubbed(self):
+        self.assertFileCollected('/etc/glance/glance-api.conf')
+
+        keys_to_mask = [
+            '4LzVZZLsPTrKw7c49ZdxqmJ8mkcWkSSnPJzgYyZWPMkrVzyB62CyCzzKVRnBS4Mz',
+            'cbVtRMqPznc4M7XKVG73yPn5fBX4NXYSC4bMb7PGYTL5RkWXnc8ZsZ9hrJy2hynf',
+            'rM5hSHsw8JSV2mJVkcszMBxtVsHbG4MW',
+        ]
+
+        for key in keys_to_mask:
+            self.assertFileNotHasContent('/etc/glance/glance-api.conf', key)
+
+    def test_heat_conf_collected(self):
+        self.assertFileCollected('/etc/glance/heat.conf')
+
+    def test_keystone_conf_collected_and_scrubbed(self):
+        self.assertFileCollected('/etc/keystone/keystone.conf')
+        self.assertFileCollected('/etc/keystone/keystone.policy.yaml')
+
+        self.assertFileNotHasContent(
+            '/etc/keystone/keystone.conf', '2zx9jZZtxdn4grG3xcMV4PwgGwY7X7fP')
+
+    def test_keystone_ldap_conf_scrubbed(self):
+
```

Comment:
Nitpick: redundant empty line?

Reply:
Yup, there's probably a few of these, will clean up once I have a full set of tests passing